### DISCUSSION TITLE
Need help integrating REST apis with emqx broker

### DISCUSSION THREAD

**Discussion by ankurchaudhary627** [emqx/emqx#12022] [Q&A] [answered]
Hi guys,

I am using version - 4.0.2 of emqx broker running inside docker container.

I need to enable and integrate rest apis provided by emqx to be able to programmatically perform user creation for client, list users, etc in my python project.

Please help me what should be the port, endpoint, plugin needed to do the same. I tried to search online and ask gpt but since the version I use is bit old, didn't get any relevant info.

Thanks..!!

**Comment by ieQu1** [comment]
Hello.

Version 4.0.2 is very old, and we don't support it anymore. Is there a specific reason why you chose it?

**Reply by ankurchaudhary627** [reply]
We have legacy codebase using the same version, need to integrate rest apis with it. Any help on how to do that would be great!!

**Reply by ieQu1** [reply]
If you don't use custom plugins implemented in Erlang or lua, it should be relatively easy to migrate. Newer versions include OpenAPI (Swagger) schema for the REST API, so it should help with the original question.

**Reply by ankurchaudhary627** [reply]
Due to time constraints, i need to make this thing working asap. Can you please let me know if the v4.0.2 has restful API support, because as per the [docs](https://www.emqx.io/docs/en/v4.0/advanced/auth-username.html#manage-authentication-data-using-http-api) I can see it does, but when i try to hit the same from local i get 401 unauthorized error.

**Reply by ankurchaudhary627** [reply, marked as answer]
Hey @ieQu1, got this resolved. 
The url with port to be used is - `http://localhost:18083/api/v4/auth_username`
Should use basic auth as Authorization method with the same creds as you use for dashboard login.
For the above api to work, you must have `emqx_auth_username` plugin enabled.
